auth-service/src/main/java/com/auth/authservice/datalayer/user/User.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Created by IntelliJ IDEA.
+ * <p>
+ * User: @Fube
+ * Date: 21/09/21
+ * Ticket: feat(AUTH-CPC-95)
+ * <p>
+ * User: @MaxGrabs
+ * Date: 26/09/21
+ * Ticket: feat(AUTH-CPC-13)
+ * <p>
+ * User: @Trilikin21
+ * Date: 24/09/21
+ * Ticket: feat(AUTH-CPC-64)
+ * <p>
+ * User: @JordanAlbayrak
+ * Date: 24/09/21
+ * Ticket: feat(AUTH-CPC-102)
+ * <p>
+ * User: @Fube
+ * Date: 10/10/21
+ * Ticket: feat(AUTH-CPC-357)
+ * <p>
+ * User: @Fube
+ * Date: 2021-10-14
+ * Ticket: feat(AUTH-CPC-388)
+ */
+package com.auth.authservice.datalayer.user;
+
+import com.auth.authservice.datalayer.roles.Role;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import jakarta.persistence.*;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import lombok.*;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import static java.lang.String.format;
+
+@Table(name = "users")
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder(toBuilder = true)
+public class User implements UserDetails {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @NotEmpty
+    private String username;
+
+    @NotEmpty
+    private String password;
+
+    @NotEmpty
+    @Email(message = "Email must be valid")
+    private String email;
+
+    private boolean verified;
+
+    public User(String username, String password, String email) {
+        this.username = username;
+        this.password = password;
+        this.email = email;
+    }
+
+    public User(String username, String password, String email, Set<Role> roles) {
+        this.username = username;
+        this.password = password;
+        this.email = email;
+        this.roles = roles;
+    }
+
+    @ManyToMany(fetch = FetchType.EAGER)
+    @JoinTable(
+            name = "users_roles",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id")
+    )
+    private Set<Role> roles;
+
+    @Override
+    @JsonIgnore
+    public Collection<? extends GrantedAuthority> getAuthorities() {
```

Comment:
I'm confused about how the code works for getAuthorities. What is Hashset, GrantedAuthority and "Role_%s"

Reply:
Hash set is a hash map but as a set meaning that no duplicate value can be in the Hash map.  And role_%s is simply for formatting the string.